democlient/democlient.cpp

```diff
@@ -59,8 +59,9 @@ class CppcheckExecutor : public ErrorLogger {
         cppcheck.settings().certainty.enable(Certainty::inconclusive);
     }
 
-    void run(const char code[]) {
-        cppcheck.check(FileWithDetails("test.cpp"), code);
+    template<std::size_t size>
+    void run(const char (&code)[size]) {
+        cppcheck.check(FileWithDetails("test.cpp"), reinterpret_cast<const uint8_t*>(code), size-1);
```

Comment:
the buffer that we pass to this function is always 4000 bytes right? But the actual code is much shorter typically. I think it would make sense to pass only the data until the terminating zero to simplecpp.

Reply:
Yes, that definitely felt off when I took a final look before I made it ready for review.

Reply:
This actually changes the behavior since before it was doing implicitly doing `strlen()` on the input and now we provide a fixed size.

Reply:
Fixed.